runtime-modules/council/src/lib.rs

```diff
@@ -579,13 +717,49 @@ impl<T: Trait> ReferendumConnection<T> for Module<T> {
     }
 }
 
+/////////////////// Calculations ///////////////////////////////////////////////
+
+struct Calculations<T: Trait> {
+    _dummy: PhantomData<T>, // 0-sized data meant only to bound generic parameters
+}
+
+impl<T: Trait> Calculations<T> {
+    /// Calculate current reward for the recipient.
+    fn get_current_reward(
+        council_member: &CouncilMemberOf<T>,
+        reward_per_block: Balance<T>,
+        now: T::BlockNumber,
+    ) -> Balance<T> {
+        council_member.unpaid_reward
+            + T::blocks_to_balance(&(now - council_member.last_payment_block)) * reward_per_block
+    }
+
+    /// Retrieve current budget's balance and calculate missing balance for reward payment.
+    fn payable_reward(
+        budget_balance: &Balance<T>,
+        reward_amount: &Balance<T>,
+    ) -> (Balance<T>, Balance<T>) {
+        // check if reward has enough balance
+        if reward_amount <= budget_balance {
+            return (*reward_amount, 0.into());
+        }
+
+        // calculate missing balance
+        let missing_balance = *reward_amount - *budget_balance;
```

Comment:
Is this safe, I think perhaps saturated arithmetic is required, this applies to a few other places also. Please consult @shamil-gadelshin about this.

Reply:
Done